File: Eunleelee/2024-02/2월 18일/의상.py
from collections import defaultdict

def solution(clothes):
    answer = 1
    dic = defaultdict(list)
    for i in clothes:
        dic[i[1]].append(i[0])
    for k in dic:
        answer*=len(dic[k]) + 1
    answer -= 1
    #print(dic)
    return answer

File: wooseongshin/2월/18일/의상.java
import java.util.*;

class Solution {
    public int solution(String[][] clothes) {
        int answer = 1;
        HashMap<String, Integer> hm = new HashMap<>();
        for (int i = 0; i < clothes.length; i++) {
            hm.put(clothes[i][1], hm.getOrDefault(clothes[i][1], 0) + 1);
        }
        for (String key : hm.keySet()) {
            answer *= (hm.get(key) + 1);
        }
        answer -= 1;
        return answer;
    }
}
